scio-extra/src/main/scala/com/spotify/scio/extra/csv/syntax/SCollectionSyntax.scala

```diff
@@ -61,5 +61,11 @@ trait SCollectionSyntax {
       self
         .withName("Read CSV")
         .applyTransform(ParDo.of(CsvIO.ReadDoFn[T](csvConfiguration)))
+
+    def readCsvWithFilename[T: HeaderDecoder: Coder](csvConfiguration: CsvConfiguration)
```

Comment:
CSVs are not splittable due to quoting and multiline values, but also they _may_ have a header in which case reading the whole file is required anyway